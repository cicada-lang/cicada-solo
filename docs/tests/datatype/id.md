---
title: Id
---

# Id

``` cicada
datatype Id(T: Type) (x: T, y: T) {
  my_refl(x: T): Id(T, x, x)
}
```

``` cicada
Id.my_refl
check! Id.my_refl(1): Id(Nat, 1, 1)
```
