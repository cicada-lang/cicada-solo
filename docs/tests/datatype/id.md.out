(vague T, x) => Id.my_refl(vague T, x): (vague T: Type, x: T) -> Id(T, x, x)
Id.my_refl(vague String, "1"): Id(String, "a", "a")